src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -561,9 +564,9 @@ public Optional<FieldChange> setField(Field field, String value, EntriesEventSou
         fields.put(field, value.intern());
 
         FieldChange change = new FieldChange(this, field, oldValue, value);
-        if (isNewField) {
+        if (isNewField && eventSource != null) {
```

Comment:
When did it happen that `eventSource` was null?

All usages of the old code show that a non-null parameter was passed to this method:

![grafik](https://user-images.githubusercontent.com/1366654/200424819-b5ef9986-0238-41af-9709-f8f9ff38ef09.png)

Reply:
Oh, I was testing why that `library was modified by external program` bug (issue #9187) was happening, so I tried manually setting `eventSource` to null, as there were `Objects.requireNonNull()` calls to the other parameters, but not for `eventSource`, so I assumed it be null. However, an exception on upper stack was thrown, so I assumed this value can't actually be null and modified the code to check for this.

As this seems to be old code, should I just leave how it was before? I could also just include  a `Objects.requireNonNull()` for `eventSource` as well...

Reply:
+1 for "leave how it was before" and adding `Objects.requireNonNull`

Reply:
Ok, done so.